src/lib/comms/sol-http-server-impl-microhttpd.c

```diff
@@ -293,10 +319,29 @@ http_server_handler(void *data, struct MHD_Connection *connection, const char *u
         return MHD_YES;
     }
 
-end:
+    SOL_PTR_VECTOR_FOREACH_IDX (&server->dirs, dir, i) {
+        struct stat st;
+        fd = get_static_file(dir, url);
```

Comment:
Might be a good idea to set `status` to "Access denied" on `EACCESS` and "Unavailable" on ENFILE/EMFILE.